snap/validate.go

```diff
@@ -727,6 +727,50 @@ func (layout *Layout) constraint() LayoutConstraint {
 	return mountedTree(path)
 }
 
+// layoutRejectionList contains directories that cannot be used as layout
+// targets.  Nothing there, or underneath can be replaced with $SNAP or
+// $SNAP_DATA content, even from the point of view of a single snap.
```

Comment:
also SNAP_COMMON. Perhaps also mention SNAP_USER_COMMON and SNAP_USER_DATA in some capacity?